packages/next/build/webpack-config.ts

```diff
@@ -88,6 +88,22 @@ const devtoolRevertWarning = execOnce(
 
 let loggedSwcDisabled = false
 let loggedIgnoredCompilerOptions = false
+let swcTraceFlushGuard: unknown = null
+
+/**
+ * Teardown swc's trace subscriber if there's an initialized flush guard exists.
+ *
+ * This is workaround to amend behavior with process.exit
+ * (https://github.com/vercel/next.js/blob/4db8c49cc31e4fc182391fae6903fb5ef4e8c66e/packages/next/bin/next.ts#L134=)
+ * seems preventing napi's cleanup hook execution (https://github.com/swc-project/swc/blob/main/crates/node/src/util.rs#L48-L51=),
+ *
+ * instead parent process manually drops guard when process gets signal to exit.
+ */
+process.on('exit', () => {
```

Comment:
This doesn't seem to be the best place to add this 🤔 I guess you might want to look at how cleanup works for .next/trace

Reply:
I could see next/trace have flushAllTraces, but it doesn't seem to be wired to process eventhandler like this. Mind elaborate some suggestions?

The reason I placed this in here is we have to pass around reference to `flushGuard` and we can isolate in this module without passing flushGuard in some way.